pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -103,7 +109,22 @@ func (nc *nodeConfig) configureNetwork() error {
 		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlayMac,
 			nc.node.GetName()))
 	}
+	return nil
+}
+
+// getInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func getInstanceID(providerID string) string {
+	providerTokens := strings.Split(providerID, "/")
+	return providerTokens[len(providerTokens)-1]
+}
 
+// applyWorkerLabel applies the worker label to the Windows Node we created.
+func (nc *nodeConfig) applyWorkerLabel() error {
+	nc.node.Labels[WorkerLabel] = ""
+	if _, err := nc.k8sclientset.CoreV1().Nodes().Update(nc.node); err != nil {
+		return errors.Wrap(err, "error applying worker label to node object")
+	}
```

Comment:
Don't you need to update `nc.node` after the `Update` call with a `Get`? Otherwise it can be stale.

Reply:
I am assuming the `Update()` call returns the latest object

Reply:
yes